hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -378,6 +380,25 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		})
 	}
 
+	// Reconcile the HostedControlPlane audit webhook config if specified
+	// reference from the HostedCluster and syncing the secret in the control plane namespace.
+	{
+		if len(hcluster.Spec.AuditWebhook.Name) > 0 {
+			var src corev1.Secret
+			if err := r.Client.Get(ctx, ctrlclient.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.AuditWebhook.Name}, &src); err != nil {
+				return ctrl.Result{}, fmt.Errorf("failed to get audit webhook config %s: %w", hcluster.Spec.AuditWebhook.Name, err)
+			}
+			configData, ok := src.Data[kas.AuditWebhookKubeconfigKey]
+			if !ok {
+				return ctrl.Result{}, fmt.Errorf("audit webhook secret does not contain key %s", kas.AuditWebhookKubeconfigKey)
+			}
+			hostedControlPlaneAuditWebhookSecret := kasmanifests.KASAuditWebhookConfigFile(controlPlaneNamespace.Name)
+			_, err = controllerutil.CreateOrUpdate(ctx, r.Client, hostedControlPlaneAuditWebhookSecret, func() error {
+				return reconcileAuditWebhook(hostedControlPlaneAuditWebhookSecret, configData)
```

Comment:
nit: function is trivial enough to inline IMO

Reply:
done